### DISCUSSION TITLE
Misorientation between grains

### DISCUSSION THREAD

**Discussion by sushdash** [neperfepx/neper#460] [Q&A] [answered]
Hello!
How to calculate the misorientation angles between grains after deformation through post-processing?

**Comment by rquey** [comment, marked as answer]
I've just added `-statelset2d theta`, which will give you the disorientation angles of all grain boundaries. You may also want to use `-statelt2d theta` for element-scale information.

**Reply by sushdash** [reply]
By using this, I could find the disorientation of neper generated microstructure which is undeformed. But how to find the same for the deformed one?

**Reply by rquey** [reply]
Good point.

Ideally, we would do it in -S and the simulation directory, but we are not there yet.

So, I've added options `-transform "node(<file_name>)"` and `-transform "ori(<file_name>)"` to -M, which override the node coordinates and elset or element orientations of a mesh, respectively. You can use them to "update" the mesh with the values you get after deformation (and which are available in the simulation directory):

```
neper -M -loadmesh undeformed.msh -transform "node(<file_name>),ori(<file_name>)" -o deformed
```

Of note, if you load a simulation directory in -V, you should be able to color grain boundaries by disorientation angle using `-dataelset2d theta` and `-dataelt2d theta`, assuming this is something you want to do.

Version `4.4.2-14`.
